### PR TITLE
chore: Improve performance of table inline editing

### PR DESCRIPTION
**Pull request by timogasda**
### Description
When a table has editable cells, it renders a few more elements in every cell to make them clickable and editable. These additional elements can cause performance regressions when compared to a table that doesn't have inline editing enabled.

This change attempts to improve the performance of editable tables by removing the amount and complexity of additional elements:

* The `<Button>` is replaced with a `<button>`
* The edit icon is rendered conditionally when the cell is hovered or focus

Related links, issue #, if available: Dry run build 6431111657.

### How has this been tested?

This table contains some explorations to improve the rendering performance of tables with inline editing.

The measurements use a Cloudscape table with 100 columns and 100 rows, with all cells being editable. First time render performance is measured in production mode and React 16.14. The browser is Chrome on an M1 Pro MacBook.

The time to render is measured by measuring the time between the requested table render and the next `requestAnimationFrame`.

| Theme | Edit button | Edit icon | Time to render (ms) | Compared to current | Compared to no editing | Notes |
| - | - | - | - | - | - | - |
| VR | `<InternalButton>` | `<InternalButton>` | 1646.4 | 0.00% | 275.89% | Current implementation |
| VR | `<span role=button>` | `<InternalIcon>` | 1358 | -17.52% | 210.05% |  |
| VR | `<span role=button>` | `<SimpleIcon>` | 1254.66667 | -23.79% | 186.45% | SimpleIcon is a "lite" version of the Icon component |
| VR | `<span role=button>` | inline svg icon | 1255.8 | -23.72% | 186.71% |  |
| VR | `<span>` | ✏️ | 860.57143 | -47.73% | 96.48% |  just an experiment |
| VR | `<span role=button>` | conditional `<InternalIcon>` | 765.8 | -53.49% | 74.84% | Only show icon on hover/focus. **Proposed solution.** | 
| Classic | No inline editing |  | 451.5 | -72.58% | 0.00% |  | Reference table with no inline editing |
| VR | No inline editing |  | 438 | -73.40% | 0.00% |  | 

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/table/body-cell/index.tsx [hunk 1]
```diff
@@ -70,15 +69,14 @@ function TableCellEditable<ItemType>({
...
-              variant="inline-icon"
-            />
+          <span
+            className={styles['body-cell-editor']}
+            aria-label={ariaLabels?.activateEditLabel?.(column)}
+            role="button"
+            tabIndex={0}
+            ref={editActivateRef}
+          >
+            <Icon name="edit" />
```
Comment (timogasda) [inline src/table/body-cell/index.tsx:79]: Todo: Active and hover color for the icon.

**Comment by gethinwebster** [comment]
Keyboard interactions seem to be broken: I can't activate edit on enter (or space), whereas I could previously

**Comment by just-boris** [comment]
Also this PR makes significant enough changes to make a dry-run to live (to reveal if anything relied on `<button>` in their selectors, same as our tests did)

**Comment by timogasda** [comment]
@gethinwebster @just-boris Sorry, I probably should have kept this PR in Draft a little longer as I'm still finalizing it. Will let you know once it's ready :)

File: pages/table/performance.page.tsx
Comment (timogasda) [inline pages/table/performance.page.tsx:1]: I'd be curious to hear from the reviewer(s) if this page would be useful to have in the code base.

File: src/table/body-cell/index.tsx [hunk 2]
```diff
@@ -58,6 +62,8 @@ function TableCellEditable<ItemType>({
         isVisualRefresh && styles['is-visual-refresh']
       )}
       onClick={!isEditing ? onEditStart : undefined}
+      onMouseEnter={() => setHasHover(true)}
```
Comment (gethinwebster) [inline src/table/body-cell/index.tsx:65]: [minor, possibly premature over-optimization] is there a performance impact of these not using `useCallback` and therefore creating new instances on each render?

**Review by gethinwebster — APPROVED** [review]

**Review by just-boris — APPROVED** [review]

**Pull request merged**